src/utils/events/event-manager.js

```diff
@@ -163,7 +186,6 @@ export default class EventManager {
       };
 
       // Calculate center relative to the root element
-      // TODO/xiaoji - avoid using getBoundingClientRect for perf?
```

Comment:
May not be conclusive. It reportedly depends on the surrounding DOM, how deep we are in the DOM tree, styling etc.

Whether it is worth the trouble to avoid I don't know...